xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/MultiUserChatServiceImpl.java

```diff
@@ -319,6 +344,8 @@ public MultiUserChatServiceImpl(final String subdomain, final String description
 
         localUsers = new ConcurrentHashMap<>();
         localMUCRoomManager = new LocalMUCRoomManager(this);
+
+        ClusterManager.addListener(this);
```

Comment:
I put this change in some time ago. Don't know if it is still relevant. Possibly all the handling of cluster events can be done in other classes now.